### PR TITLE
Fix for #67423 ansible issue

### PR DESCRIPTION
**Pull request by Vaishnavi-infoblox**
##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by Vaishnavi-infoblox** [comment]
Fix for #67423

**Comment by megabreit** [comment]
@Vaishnavi-infoblox You added only 5 number options but there are 8 name options. Just want to make sure this is no typo...

File: lib/ansible/modules/net_tools/nios/nios_fixed_address.py [hunk 1]
```diff
@@ -189,13 +189,18 @@ def options(module):
     special_options = ['routers', 'router-templates', 'domain-name-servers',
                        'domain-name', 'broadcast-address', 'broadcast-address-offset',
                        'dhcp-lease-time', 'dhcp6.name-servers']
+    special_num =  [3, 6 , 15, 28, 51]
```
Comment (Akasurde) [inline lib/ansible/modules/net_tools/nios/nios_fixed_address.py]:
```suggestion
    special_num =  [3, 6, 15, 28, 51]
```

File: lib/ansible/modules/net_tools/nios/nios_fixed_address.py [hunk 2]
```diff
@@ -189,13 +189,18 @@ def options(module):
...
     options = list()
     for item in module.params['options']:
         opt = dict([(k, v) for k, v in iteritems(item) if v is not None])
         if 'name' not in opt and 'num' not in opt:
             module.fail_json(msg='one of `name` or `num` is required for option value')
-        if opt['name'] not in special_options:
-            del opt['use_option']
+        if 'name' in opt:
```
Comment (Akasurde) [inline lib/ansible/modules/net_tools/nios/nios_fixed_address.py]:
```suggestion
        if 'name' in opt and opt['name'] not in special_options:
```

File: lib/ansible/modules/net_tools/nios/nios_fixed_address.py [hunk 3]
```diff
@@ -189,13 +189,18 @@ def options(module):
...
+            if opt['name'] not in special_options:
+                del opt['use_option']
+        if 'num' in opt:
```
Comment (Akasurde) [inline lib/ansible/modules/net_tools/nios/nios_fixed_address.py]:
```suggestion
        if 'num' in opt and opt['num'] not in special_num:
```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Please add correct title and description so that it will convey why this change is made in first place.

**Comment by Vaishnavi-infoblox** [comment]
> @Vaishnavi-infoblox You added only 5 number options but there are 8 name options. Just want to make sure this is no typo...

options-router-templates, broadcast-address-offset, dhcp6.name-servers don't have any associated number

**Pull request closed without merging**